### ISSUE TITLE
Add area under the Precision-Recall curve (AUPRC): a useful metric  for class imbalance.

### ISSUE THREAD

**Issue by gm039** [pycaret/pycaret#806]
The area under the Precision-Recall curve (AUPRC) is very useful metric for class imbalance. Here is the source (https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html).

I tried using the add_metric feature in Pycaret 2.2 as below.

_from sklearn.metrics import average_precision_score
add_metric('AUPRC_ID','AUC_PRC',average_precision_score, greater_is_better = True)_

But the scores are different from the score obtained using the evaluate_model(tuned_model_best) precision-recall curve (See the snapshot below).

![image](https://user-images.githubusercontent.com/50217085/98030946-c54cf680-1ddf-11eb-9950-df799ac3edcd.png)

Additionally, the AUC_PRC score seems to takes Target and Predicted Label as input argument instead of Target and Predicted Score.

**Comment by gm039** [comment]
@pycaret What is the default input argument to the metric defined through the add_metric feature in Pycaret 2.2?

**Comment by pycaret** [comment]
@gm039 You can actually access the metrics PyCaret is using by calling the `get_metrics` function.

**Comment by gm039** [comment]
@pycaret Thanks! Below are the metrics used.

![image](https://user-images.githubusercontent.com/50217085/98170103-6e1c5400-1ebb-11eb-82c1-717aa3d15ea6.png)

The input argument to the metric should be Target and Predicted Score but its taking Target and Predicted Label as  verified in the previous snapshot.

Unfortunately, the average precision calculated is different from the score obtained using the evaluate_model(tuned_model_best) precision-recall curve (See the previous snapshot's PR curve).

**Comment by pycaret** [comment]
@gm039 You can pass the `target = 'pred_proba'` inside the `add_metric` call and it will work just fine :) See below:

```
from pycaret.datasets import get_data
data = get_data('juice')

from pycaret.classification import *
s = setup(data, target = 'Purchase', session_id = 123, silent = True)

from sklearn.metrics import average_precision_score
add_metric('apc', 'APC', average_precision_score, target = 'pred_proba')

lr = create_model('lr')

predict_model(lr);

plot_model(lr, plot = 'pr')
```
![image](https://user-images.githubusercontent.com/58118658/98171305-83927d80-1ebd-11eb-8fe1-2530a21d6632.png)

![image](https://user-images.githubusercontent.com/58118658/98171311-88efc800-1ebd-11eb-90e4-2ec599679082.png)

Hope this helps?

**Comment by pycaret** [comment]
@gm039 Please close the issue, if this answers the question.

**Comment by gm039** [comment]
@pycaret Thankyou so much. It worked.
